tests/test_settings.py

```diff
@@ -55,6 +56,18 @@ def test_settings_transforms_repository_config(tmpdir):
     assert s.disable_progress_bar is False
 
 
+@pytest.mark.parametrize("verbose", [True, False])
+def test_setup_logging(verbose: bool):
```

Comment:
Instead of the `if` (which I'm wary of in tests), let's use the expected log level as a parameter

```suggestion
@pytest.mark.parametrize("verbose, log_level", [(True, logging.INFO), (False, logging.WARNING])
def test_setup_logging(verbose, log_level):
```

Also, note that our tests are not type-checked. In general, I avoid adding type annotations to un-checked code, because they could be wrong, which would be confusing ala incorrect comments.